[skip ci] Update the contributing guide to add alerts (#19537)
tachiyomiorg/tachiyomi-extensions: CONTRIBUTING.md

If you find yourself needing additional functionality, you can add more dependencies to your `build.gradle` file.
Many of [the dependencies](https://github.com/tachiyomiorg/tachiyomi/blob/master/app/build.gradle.kts) from the main Tachiyomi app are exposed to extensions by default.

> [!NOTE]

> > Several dependencies are already exposed to all extensions via Gradle version catalog.

Reviewer: ```suggestion
> Several dependencies are already exposed to all extensions via Gradle's version catalog.
```
Sounds like it should be "a version catalog", not "the version catalog"

> > To view which are available check the `libs.versions.toml` file under the `gradle` folder.

Reviewer: ```suggestion
> To view which are available check the `gradle/libs.versions.toml` file.
```

$ adb shell am start -d "<your-link>" -a android.intent.action.VIEW
```

> [!CAUTION]

> > The activity has no support to any Kotlin Intrinsics specific methods or calls,
> > and using it will causes crashes in the activity. Consider using Java's equivalent
> > methods instead, such as using `String`'s `equals()` instead of using `==`.

Reviewer: ```suggestion
> The activity does not support any Kotlin Intrinsics specific methods or calls,
> and using them will causes crashes in the activity. Consider using Java's equivalent
> methods instead, such as `String`'s `equals()` instead of `==`.
```

Then the class name and the `name` attribute value can be changed. Also don't forget to update the extension name and class name in the individual Gradle file if it is not a multisrc extension.

> [!IMPORTANT]
> The package name **needs** to be the same (even if it has the old name), otherwise users will not receive the extension update when it gets published in the repository. If you're changing the name of a multisrc source, you can manually set it in the generator class of the theme by using `pkgName = "oldpackagename"`.

The `id` also needs to be explicity set to the old value if you're changing the `lang` attribute.

> [!NOTE]

> > If the source has also changed their theme and hence making the old saved URLs
> > in the library incompatible, you can skip this part and just rename by changing
> > the `name` in the source class and in the Gradle file, but keeping the old package.
> > By doing so, a new `id` will be generated so users will be forced to migrate to
> > update their entries' URLs.

Reviewer: ```suggestion
> If the source has also changed their theme you can instead just change
> the `name` field in the source class and in the Gradle file. By doing so
> a new `id` will be generated and users will be forced to migrate.
```
*might be throwing away too much information, but the details aren't really that important here, IMHO

-W -S -n eu.kanade.tachiyomi/eu.kanade.tachiyomi.ui.main.MainActivity -a eu.kanade.tachiyomi.SHOW_CATALOGUES
```

> [!IMPORTANT]

> > If you're deploying to Android 11 or higher, enable the "Always install with package manager" option in the run configurations, otherwise you may face issues such as building the extension without your new modifications.

Reviewer: ```suggestion
> If you're deploying to Android 11 or higher, enable the "Always install with package manager" option in the run configurations. Without this option enabled, changes will not apply until you restart Tachiyomi.
```

Author: That's not really accurate. If the option isn't marked, what could happen is Android Studio compiling it but when you run it doesn't have the changes in the extension which you might have done.

Reviewer: Not sure how you would word it then.

As an anecdote, I have observed the described issue, and solved it by restarting Tachiyomi. I did not have to recompile it for it to start working, kinda like the app was still holding on to a reference to the old version.

show up in the [`Logcat`](https://developer.android.com/studio/debug/am-logcat) panel of Android Studio.

### Inspecting network calls

One of the easiest way to inspect network issues (such as HTTP errors 404, 429, no chapter found etc.) is to use the [`Logcat`](https://developer.android.com/studio/debug/am-logcat) panel of Android Studio and filtering by the `OkHttpClient` tag.

To be able to check the calls done by OkHttp, you need to enable verbose logging in the app, that is not enabled by default and is only included in the Preview versions of Tachiyomi. To enable it, go to More -> Settings -> Advanced -> Verbose logging. After enabling it, don't forget to restart the app.

> Inspecting the Logcat allows you to get a good look at the call flow and it's more than enough in most cases where issues occurs. However, alternatively, you can also use an external tool like `mitm-proxy`. For that, refer to the next sections.

Reviewer: ```suggestion
Inspecting the Logcat allows you to get a good look at the call flow and it's more than enough in most cases where issues occurs. However, alternatively, you can also use an external tool like `mitm-proxy`. For that, refer to the subsequent sections.
```

> On newer Android Studio versions, you can use its built-in Network Inspector inside the
> App Inspection tool window, which provides a nice GUI to inspect the requests done in the app.

Reviewer: ```suggestion
On newer Android Studio versions, you can use its built-in Network Inspector found within the
App Inspection tool window. This feature provides a nice GUI to inspect the requests made in the app.
```

> To use it, follow the [official documentation](https://developer.android.com/studio/debug/network-profiler) and select Tachiyomi package name on the process list.

Reviewer: ```suggestion
To use it, follow the [official documentation](https://developer.android.com/studio/debug/network-profiler) and select Tachiyomi package name in the process list.
```
